### ISSUE TITLE
[Bug]: aws_secretsmanager_secret_rotation switch from automatically_after_days to schedule_expression failed

### ISSUE THREAD

**Issue by CyrilDevOps** [hashicorp/terraform-provider-aws#30540]
### Terraform Core Version

1.4.4

### AWS Provider Version

4.62.0

### Affected Resource(s)

```terraform
resource "aws_secretsmanager_secret_rotation" "rotation" {

  rotation_lambda_arn = var.aurora_secrets_rotation_lambda_arn
  secret_id           = aws_secretsmanager_secret.secret.id

  rotation_rules {
    automatically_after_days = local.t_days
    schedule_expression = local.t_schedule
  }
}
```

### Expected Behavior

In the past we only use

```terraform
  rotation_rules {
    automatically_after_days = local.t_days
  }
```

Now we switch to cron style schedule with :

```terraform
  rotation_rules {
    automatically_after_days = local.t_days
    schedule_expression = local.t_schedule
  }
```

where local.t_days is null
and 
local.t_schedule = "cron(...)"

I wanted to keep the same aws_secretsmanager_secret_rotation to do an update instead of having two different resources with count for both config (days/schedule)
but during the apply I got the error : 

```shell
│ Error: updating Secrets Manager Secret Rotation "arn:aws:secretsmanager:us-west-2:123456:secret:/secrets-rQ6Iq5" : InvalidParameterException: You cannot specify both rotation frequency and schedule expression together.
```

### Actual Behavior

plan return : 

```shell
  # module.databases_secrets["xx"].module.secrets["xxxx"].aws_secretsmanager_secret_rotation.rotation[0] will be updated in-place
  ~ resource "aws_secretsmanager_secret_rotation" "rotation" {
        id                  = "arn:aws:secretsmanager:us-west-2:123456:secret:/secrets-rQ6Iq5"
        # (3 unchanged attributes hidden)

      ~ rotation_rules {
          + schedule_expression      = "cron(0 0 ? * SAT#1 *)"
            # (1 unchanged attribute hidden)
        }
    }
```

doesn't seems to catch the removal of automatically_after_days
and the apply gave me : 

```shell
│ Error: updating Secrets Manager Secret Rotation "arn:aws:secretsmanager:us-west-2:123456:secret:/secrets-rQ6Iq5" : InvalidParameterException: You cannot specify both rotation frequency and schedule expression together.
```

### Terraform Configuration Files

```terraform
resource "aws_secretsmanager_secret_rotation" "rotation" {

  rotation_lambda_arn = var.aurora_secrets_rotation_lambda_arn
  secret_id           = aws_secretsmanager_secret.secret.id

  rotation_rules {
    automatically_after_days = local.t_days
    schedule_expression = local.t_schedule
  }
}
```

base on my tfvars file, local.t_days has a value and local.t_schedule is null, or local.t_days is null and local.t_schedule as a cron style schedule

### Steps to Reproduce

moving from a days config (aws provider <4.62.0)

```terraform
resource "aws_secretsmanager_secret_rotation" "rotation" {

  rotation_lambda_arn = var.aurora_secrets_rotation_lambda_arn
  secret_id           = aws_secretsmanager_secret.secret.id

  rotation_rules {
    automatically_after_days = local.t_days
  }
}
```

to a schedule config :

```terraform
resource "aws_secretsmanager_secret_rotation" "rotation" {

  rotation_lambda_arn = var.aurora_secrets_rotation_lambda_arn
  secret_id           = aws_secretsmanager_secret.secret.id

  rotation_rules {
    automatically_after_days = local.t_days
    schedule_expression = local.t_schedule
  }
}
```

### Would you like to implement a fix?

None

**Comment by CyrilDevOps** [comment]
I wanted to avoid two resource and count for both style of config, as our config today is with days (aws < 4.62) but want to move them to cron style.
a two resource config may create 'problem' as one would be destroy and one would be created by they point to the same secrets, so if the order is wrong, I would have no rotation at all.
Wanted also to update the schedule but not force a immediate rotation. (don't know if aws allow that).

**Comment by CyrilDevOps** [comment]
Same error when using the setup :
  dynamic "rotation_rules" {
    for_each = local.t_days != null ? [1] : []
    content {
      automatically_after_days = local.t_days
    }
  }

  dynamic "rotation_rules" {
    for_each = local.t_schedule != null ? [1] : []
    content {
      schedule_expression = local.t_schedule
    }
  }

**Comment by CyrilDevOps** [comment]
Same problem when I try to change an existing schedule_expression :
  # module.databases_secrets["xxxx"].module.root.aws_secretsmanager_secret_rotation.rotation_schedule[0] will be updated in-place
      ~ rotation_rules {
          ~ schedule_expression      = "cron(0 0 ? * SAT#1 *)" -> "cron(0 0 ? * SUN#1 *)"
            # (1 unchanged attribute hidden)
        }

│ Error: updating Secrets Manager Secret Rotation "arn:aws:secretsmanager:us-east-1:xxxx:secret:/monsecret-123456" : InvalidParameterException: You cannot specify both rotation frequency and schedule expression together.

**Comment by joaocfernandes** [comment]
> Same problem when I try to change an existing schedule_expression :
> 
> ```shell
>   # module.databases_secrets["xxxx"].module.root.aws_secretsmanager_secret_rotation.rotation_schedule[0] will be updated in-place
>       ~ rotation_rules {
>           ~ schedule_expression      = "cron(0 0 ? * SAT#1 *)" -> "cron(0 0 ? * SUN#1 *)"
>             # (1 unchanged attribute hidden)
>         }
> 
> │ Error: updating Secrets Manager Secret Rotation "arn:aws:secretsmanager:us-east-1:xxxx:secret:/monsecret-123456" : InvalidParameterException: You cannot specify both rotation frequency and schedule expression together.
> ```

I am also affected by this.


```
  rotation_rules {
    schedule_expression = "cron(0 14 1/2 * ? *)"
  }
```

Plan result:

```

      ~ rotation_rules {
          ~ schedule_expression      = "cron(0 18 1/2 * ? *)" -> "cron(0 14 1/2 * ? *)"
            # (1 unchanged attribute hidden)
        }

````


Apply result:


```
Error: updating Secrets Manager Secret Rotation "arn:aws:secretsmanager:region:xyzw:secret:/mysecret/credentials-xyzw" :

InvalidParameterException: You cannot specify both rotation frequency and schedule expression together.
```

**Comment by tudoramariei** [comment]
This seems to be a duplicate of #28016
The feature [has been made available in 4.62.0](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md#4620-april--6-2023)
